apps/vscode/command_client/command_client.py

```diff
@@ -197,20 +197,34 @@ def run_command(
     return decoded_contents["returnValue"]
 
 
+command_server_directory = None
+
+
+# NB: See https://github.com/talonhub/community/issues/966 for why we do OS-specific temp dirs
+def get_platform_specific_communication_dir_path():
+    home_dir = Path(os.path.expanduser("~"))
+    if app.platform == "linux" or app.platform == "mac":
+        return (
+            Path(home_dir) / f".talon/.comms/{actions.user.command_server_directory()}"
```

Comment:
I thought accessing a hidden folder was also a problem for snaps, no? Cc/ @auscompgeek

Reply:
This is true for most snaps by default iirc, but afaik vscode uses something called classic confinement, which has unrestricted access to files. This makes sense since it's an IDE and users would expected to be able to access anything as their user anyway. So I think we should be okay here (though I haven't tested with the snap yet).